prov/tcp/src/tcpx_ep.c

```diff
@@ -747,3 +825,46 @@ int tcpx_passive_ep(struct fid_fabric *fabric, struct fi_info *info,
 	free(_pep);
 	return ret;
 }
+
+int tcpx_set_port_range (struct tcpx_pep *pep)
+{
+	char *low_range;
+	char *high_range;
+	int  high = 0;
+	int  low  = 0;
+
+	if (!(fi_param_get_str(&tcpx_prov, "port_low_range", &low_range))) {
+		low = atoi(low_range);
+	}
+
+	if (!(fi_param_get_str(&tcpx_prov, "port_high_range", &high_range))) {
+		high = atoi(high_range);
+	}
+
+	if (low > TCPX_PORT_MAX_RANGE || low <= 0) {
+		low = 0;
+	}
+
+	if (high > TCPX_PORT_MAX_RANGE || high <= 0) {
+		high = 0;
+	}
+
+	if (high != 0 && low == 0) {
+		FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,"Invalid info\n");
+		return -FI_EINVAL;
+	}
+
+	if (high < low) {
+		if (high == 0 &&  high < low) {
+			high = TCPX_PORT_MAX_RANGE;
+		} else {
+			low  = 0;
+			high = 0;
+		}
+	}
```

Comment:
Discard all invalid input
```
if (low < 0 || high < 0 || low > high) {
    FI_WARN(....);
    return;
}
```
I'd be fine with this, however:
```
if (high > TCP_PORT_MAX)
    high = TCP_PORT_MAX;
```
This way, the user doesn't need to define low at all, or know what the true maximum port value is.  The resulting range of values will still be correct.